runtime/gc_vlhgc/IncrementalGenerationalGC.cpp

```diff
@@ -2254,13 +2255,15 @@ MM_IncrementalGenerationalGC::exportStats(MM_EnvironmentVLHGC *env, MM_Collectio
 				if (classesPotentiallyUnloaded && !isMarked((J9Object *)spine)) {
 					stats->_arrayletUnknownLeaves += 1;
 					/* is this first arraylet leaf? */
-					if (region->getLowAddress() == mmPointerFromToken((J9VMThread*)env->getLanguageVMThread(), _extensions->indexableObjectModel.getArrayoidPointer(spine)[0])) {
+					fj9object_t firstArrayletLeaf = GC_SlotObject::readSlot(_extensions->indexableObjectModel.getArrayoidPointer(spine), compressed);
+					if (region->getLowAddress() == mmPointerFromToken((J9VMThread*)env->getLanguageVMThread(), firstArrayletLeaf)) {
```

Comment:
These 2 lines could be more concise by using GC_SlotObject::readReferenceFromSlot() instead of readSlot&mmPointerFromToken

Reply:
Creating SlotObject requires and OMR VM, which is not in evidence here - is it worth the work to fetch that?

Reply:
Actualyl, we have a javaVM locally, so I the dig isn't very far.

Reply:
Thanks.

BTW, we should probably retire mmPointerFromToken macro at some point. Seems like it's used at only two more spots (in ObjectAccessBarrier code), where again readReferenceFromSlot could be used, instead.

Reply:
Done. These are no longer used:
```
extern "C" mm_j9object_t j9gc_objaccess_pointerFromToken(J9VMThread *vmThread, fj9object_t token);
extern "C" fj9object_t j9gc_objaccess_tokenFromPointer(J9VMThread *vmThread, mm_j9object_t object);
```

Would you like those removed as well?

Reply:
Looks like it would be easy to go one step further and get rid of the token functions in the barriers as well (replace with SlotObject).

Reply:
I'm happy with the changes relevant to RemSet. Widening the scope of changes that re not relevant is up to you.